src/shared/filters/api-exception.filter.ts

```diff
@@ -0,0 +1,310 @@
+import { 
+    Catch, 
+    ArgumentsHost, 
+    BadRequestException, 
+    UnauthorizedException, 
+    ForbiddenException, 
+    NotFoundException,
+    MethodNotAllowedException,
+    NotAcceptableException,
+    RequestTimeoutException,
+    ConflictException,
+    GoneException,
+    ExceptionFilter,
+    PayloadTooLargeException,
+    UnsupportedMediaTypeException,
+    UnprocessableEntityException
+} from '@nestjs/common';
+import { HttpAdapterHost } from '@nestjs/core';
+import { ApiException } from '../api-exception.model';
+
+@Catch(BadRequestException)
+class BadRequestExceptionFilter implements ExceptionFilter {
+    constructor(private readonly httpAdapterHost: HttpAdapterHost) {}
+
+    catch(exception: BadRequestException, host: ArgumentsHost): void{
+        const { httpAdapter } = this.httpAdapterHost;
+  
+        const ctx = host.switchToHttp();
+    
+        const responseBody: ApiException = {
+          statusCode: exception.getStatus(),
+          message: "Bad Request - Please check the request sent.",
+          error: exception.message,
+          timestamp: new Date().toISOString(),
+          path: httpAdapter.getRequestUrl(ctx.getRequest()),
+        };
+
+        console.log("Response to be sent :: " + responseBody)
+    
+        httpAdapter.reply(ctx.getResponse(), responseBody, exception.getStatus());
+    }
+}
+
+@Catch(UnauthorizedException)
+class UnauthorizedExceptionFilter implements ExceptionFilter {
+    constructor(private readonly httpAdapterHost: HttpAdapterHost) {}
+
+    catch(exception: BadRequestException, host: ArgumentsHost): void{
+        const { httpAdapter } = this.httpAdapterHost;
+  
+        const ctx = host.switchToHttp();
+    
+        const responseBody: ApiException = {
+          statusCode: exception.getStatus(),
+          message: "Authorization failed. Please contact the admin.",
+          error: exception.message,
+          timestamp: new Date().toISOString(),
+          path: httpAdapter.getRequestUrl(ctx.getRequest()),
+        };
+
+        console.log("Response to be sent :: " + responseBody)
+    
+        httpAdapter.reply(ctx.getResponse(), responseBody, exception.getStatus());
+    }
+}
+
+@Catch(ForbiddenException)
+class ForbiddenExceptionFilter implements ExceptionFilter {
+    constructor(private readonly httpAdapterHost: HttpAdapterHost) {}
+
+    catch(exception: BadRequestException, host: ArgumentsHost): void{
+        const { httpAdapter } = this.httpAdapterHost;
+  
+        const ctx = host.switchToHttp();
+    
+        const responseBody: ApiException = {
+          statusCode: exception.getStatus(),
+          message: "You don't have access to the requested action. Please contact the admin.",
```

Comment:
Please contact the admin. remove this in all messages.

Reply:
done